### PR TITLE
Don't serialize specific variables to the elevated Burn process

### PR DESCRIPTION
**Pull request by rseanhall**
#228 allowed the unelevated Burn process to overwrite built in variables to fix [4630](http://wixtoolset.org/issues/4630), but the elevated Burn process was using the ProgramFiles variables to make security decisions for ApprovedExeForElevation.  So mark those variables as not to be serialized to the elevated process.

### CODE REVIEW

File: src/burn/engine/variable.cpp [hunk 1]
```diff
@@ -232,37 +234,37 @@ extern "C" HRESULT VariableInitialize(
...
-        {L"ProgramFilesFolder", InitializeVariableCsidlFolder, CSIDL_PROGRAM_FILES},
+        {L"ProgramFiles64Folder", InitializeVariableRegistryFolder, CSIDL_PROGRAM_FILES, FALSE, TRUE},
+        {L"ProgramFilesFolder", InitializeVariableCsidlFolder, CSIDL_PROGRAM_FILES, FALSE, TRUE },
 #endif
-        {L"ProgramFiles6432Folder", InitializeVariable6432Folder, CSIDL_PROGRAM_FILES},
-        {L"ProgramMenuFolder", InitializeVariableCsidlFolder, CSIDL_PROGRAMS},
+        {L"ProgramFiles6432Folder", InitializeVariable6432Folder, CSIDL_PROGRAM_FILES, FALSE, TRUE},
+        {L"ProgramMenuFolder", InitializeVariableCsidlFolder, CSIDL_PROGRAMS, FALSE, TRUE},
         {L"RebootPending", InitializeVariableRebootPending, 0},
         {L"SendToFolder", InitializeVariableCsidlFolder, CSIDL_SENDTO},
```
Comment (heaths) [inline src/burn/engine/variable.cpp]: What about this one?

File: src/burn/engine/variable.cpp [hunk 2]
```diff
@@ -232,37 +234,37 @@ extern "C" HRESULT VariableInitialize(
...
-        {L"StartMenuFolder", InitializeVariableCsidlFolder, CSIDL_STARTMENU},
-        {L"StartupFolder", InitializeVariableCsidlFolder, CSIDL_STARTUP},
-        {L"SystemFolder", InitializeVariableSystemFolder, FALSE},
-        {L"System64Folder", InitializeVariableSystemFolder, TRUE},
+        {L"StartMenuFolder", InitializeVariableCsidlFolder, CSIDL_STARTMENU, FALSE, TRUE},
+        {L"StartupFolder", InitializeVariableCsidlFolder, CSIDL_STARTUP, FALSE, TRUE},
+        {L"SystemFolder", InitializeVariableSystemFolder, FALSE, FALSE, TRUE},
+        {L"System64Folder", InitializeVariableSystemFolder, TRUE, FALSE, TRUE},
         {L"SystemLanguageID", InitializeSystemLanguageID, 0},
         {L"TempFolder", InitializeVariableTempFolder, 0},
```
Comment (heaths) [inline src/burn/engine/variable.cpp]: This one.

File: src/burn/engine/variable.cpp [hunk 3]
```diff
@@ -232,37 +234,37 @@ extern "C" HRESULT VariableInitialize(
...
-        {L"TemplateFolder", InitializeVariableCsidlFolder, CSIDL_TEMPLATES},
+        {L"TemplateFolder", InitializeVariableCsidlFolder, CSIDL_TEMPLATES, FALSE, TRUE},
         {L"TerminalServer", InitializeVariableOsInfo, OS_INFO_VARIABLE_TerminalServer},
         {L"UserLanguageID", InitializeUserLanguageID, 0},
         {L"VersionMsi", InitializeVariableVersionMsi, 0},
         {L"VersionNT", InitializeVariableVersionNT, OS_INFO_VARIABLE_VersionNT},
         {L"VersionNT64", InitializeVariableVersionNT, OS_INFO_VARIABLE_VersionNT64},
-        {L"WindowsFolder", InitializeVariableCsidlFolder, CSIDL_WINDOWS},
+        {L"WindowsFolder", InitializeVariableCsidlFolder, CSIDL_WINDOWS, FALSE, TRUE},
         {L"WindowsVolume", InitializeVariableWindowsVolumeFolder, 0},
```
Comment (heaths) [inline src/burn/engine/variable.cpp]: This one.

**Comment by rseanhall** [comment]
Updated based on yesterday's feedback.  Also deleted property.h since it wasn't being used.

File: src/burn/engine/variable.cpp [hunk 4]
```diff
@@ -240,7 +242,7 @@ extern "C" HRESULT VariableInitialize(
         {L"ProgramFilesFolder", InitializeVariableCsidlFolder, CSIDL_PROGRAM_FILESX86},
 #else
         {L"ProgramFiles64Folder", InitializeVariableRegistryFolder, CSIDL_PROGRAM_FILES},
-        {L"ProgramFilesFolder", InitializeVariableCsidlFolder, CSIDL_PROGRAM_FILES},
+        {L"ProgramFilesFolder", InitializeVariableCsidlFolder, CSIDL_PROGRAM_FILES },
```
Comment (heaths) [inline src/burn/engine/variable.cpp]: Not a big deal, but cleaner if you get rid of the now-added space. Probably just if you have any other changes to make.

**Comment by heaths** [comment]
Looks good.

**Pull request merged**